### ISSUE TITLE
Add Target Company and Domain Names as Variables

### ISSUE THREAD

**Issue by riab-tam** [gophish/gophish#1396]
**Thanks for reaching out! We're happy to help resolve issues as quickly as possible.**

**Please use this template when creating a new issue. If you do not follow this template format, your issue may be closed without comment.**

Before filing a new issue, please use the search bar at the top of the browser to search for similar issues. Also, please make sure you have read our [documentation](https://getgophish.com/documentation/), which covers how to use Gophish.

What version of Gophish are you using?: 0.7.1

Brief description of the issue: Can company and domain names be added as fields for the Group Users and used in the templates as target variables?

What are you expecting to see happen? : In email templates and landing pages, be able to use something like "{{.CompanyName}}", ie. "Hello {{.FirstName}}, your boss at {{.CompanyName}} wants you to fill out a form at {{.URL}}..."

What are you seeing happen? New fields and variables; currently not an option.

If this question is related to email templates or landing pages not working as expected, please provide your template or landing page below:

```html
Insert code here
```

Please provide any terminal output that may be relevant below:

```
Insert output here
```

Please provide as many steps as you can to reproduce the problem:
 * Step 1 - 
 * Step 2 -

**Comment by bartenge** [comment]
Good afternoon
I have spend hours troubleshooting gophish, i launch campaign but it seems targeted users are not receiving mails, test with a prompt mail sent but ideally the users did receive the sent mail

**Comment by S0larflare** [comment]
I think there's definitely a case for adding in some extra variables (personally I think some custom ones which can be set somewhere) and I can see CompanyName being useful in quite a few cases. As a workaround, if you aren't using {{.Position}} for anything in particular, you could repurpose that to hold the company name you want to use and drop that in to your email template.

**Comment by quelsan** [comment]
I agree that this would be a useful feature - I've done similar things with hacky patches to Gophish, but it would be nice to have a standardized way to add arbitrary custom variables to templates/landing pages/sending profiles.

@jordan-wright : Perhaps we should look into this for #1589

**Comment by Al-TSN** [comment]
I'd like to +1 on this one.  Just figured out that we will probably repurpose {{.Position}} we have had a very hacky work around until now.  That should do the trick.  However with the uptake in spearphishing having a bunch of extra fields would be great. They don't even need to have great contextual names, e.g. {{.OtherData1}} ... {{.OtherData20}} would be great. OK, 20 may be overkill, but you never know.  @jordan-wright I'd be happy to take the same route we did with the webhooks.
